Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.h

```diff
@@ -0,0 +1,123 @@
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if PLATFORM(COCOA) && ENABLE(WEBM_EXPERIMENT)
+
+#include "MediaPlayerPrivate.h"
+#include "NotImplemented.h"
+#include <wtf/WeakPtr.h>
+#include <wtf/LoggerHelper.h>
+
+namespace WebCore {
+
+class MediaPlayerPrivateWebM : public MediaPlayerPrivateInterface, public CanMakeWeakPtr<MediaPlayerPrivateWebM>
+#if !RELEASE_LOG_DISABLED
+    , private LoggerHelper
+#endif
+{
+    WTF_MAKE_FAST_ALLOCATED;
+public:
+    MediaPlayerPrivateWebM(MediaPlayer*);
+    ~MediaPlayerPrivateWebM();
+    
+    static void registerMediaEngine(MediaEngineRegistrar);
+    
+    void load(const String&) final;
+    
+#if ENABLE(MEDIA_SOURCE)
+    void load(const URL&, const ContentType&, MediaSourcePrivateClient&) final;
+#endif
+#if ENABLE(MEDIA_STREAM)
+    void load(MediaStreamPrivate&) final { };
+#endif
+    
+    void cancelLoad() final;
+    
+    void play() final;
+    void pause() final;
+    
+    FloatSize naturalSize() const final { return m_naturalSize; };
+    
+    bool hasVideo() const final { return m_hasAudio; };
+    bool hasAudio() const final { return m_hasVideo; };
+    
+    void setPageIsVisible(bool) final;
+    
+    bool seeking() const final { return m_seeking; };
+    
+    bool paused() const final { return m_paused; };
+    
+    MediaPlayer::NetworkState networkState() const final { return m_networkState; };
+    MediaPlayer::ReadyState readyState() const final { return m_readyState; };
+    
+    std::unique_ptr<PlatformTimeRanges> buffered() const final { return makeUnique<PlatformTimeRanges>(); };
+    
+    bool didLoadingProgress() const final { return false; };
```

Comment:
I would move these into the .cpp file now as it will have to be there soon enough